crates/transaction-pool/src/validate/eth.rs

```diff
@@ -121,7 +132,10 @@ pub(crate) struct EthTransactionValidatorInner<Client, T> {
 
 // === impl EthTransactionValidatorInner ===
 
-impl<Client, Tx> EthTransactionValidatorInner<Client, Tx> {
+impl<Client, Tx> EthTransactionValidatorInner<Client, Tx>
+where
+    Client: BlockReaderIdExt,
+{
```

Comment:
I'd prefer a separate Optimism validator instead, even if this means duplicating some checks etc.

Reply:
Spoke about this offline - will be addressed in a followup PR.